__init__.py

```diff
@@ -72,17 +61,30 @@ class DeviceData(threading.Thread):
     for every new data, could work for the pH and ORP data but for the
     switches a more direct feedback is wanted."""
 
-    def __init__(self, hass, device_client: device.Device) -> None:
+    def __init__(self, hass, config) -> None:
         super().__init__()
         self.name = "Ph803wThread"
         self.hass = hass
-        self.device_client = device_client
-        self.device_client.register_callback(self.dispatcher_new_data)
-        self.device_client.register_callback(self.reset_fail_counter)
-        self.host = self.device_client.host
+        self.host = config[CONF_HOST]
+        self.device_client = None
         self._shutdown = False
         self._fails = 0
 
+    def passcode(self):
+        if self.device_client is not None:
+            return self.device_client.passcode
+        return "unknown"
+
+    def unique_name(self):
+        if self.device_client is not None:
+            return self.device_client.get_unique_name()
+        return "unknown"
+
+    def measurement(self):
+        if self.device_client is not None:
+            return self.device_client.get_latest_measurement()
+        return None
```

Comment:
These wrappers prevent sensors from erroring if the `device_client` is `None`, which can happen during initialization and reconnecting.